### PR TITLE
Fixes issue #285. Added Previews in GUI and statistics which were missing after refactoring.

### PR DESCRIPTION
**Pull request by abhishek-vinjamoori**
A few commented lines exist. Will uncomment once @anshul1912 responds.

### CODE REVIEW

**Comment by canihavesomecoffee** [comment]
@ccx-testerbot runtests master

**Comment by canihavesomecoffee** [comment]
@ccx-testerbot runtests master

**Comment by canihavesomecoffee** [comment]
@ccx-testerbot runtests master

**Comment by cfsmp3** [comment]
This only changes things related to statistics, how could it fix the issue with the preview?

**Comment by canihavesomecoffee** [comment]
@ccx-testerbot runtests master

**Comment by canihavesomecoffee** [comment]
Doesn't affect any existing tests.

File: src/ccextractor.c [hunk 1]
```diff
@@ -292,8 +230,76 @@ int main(int argc, char *argv[])
...
+				dec_ctx->frames_since_last_gop);
+			mprint ("Diff. GOP length:	   %s%+3dF",
+				print_mstime(gop_time.ms - first_gop_time.ms),
+				dec_ctx->frames_since_last_gop);
+			mprint ("	(%s)\n\n",
+				print_mstime(gop_time.ms - first_gop_time.ms
+				+(LLONG) ((dec_ctx->frames_since_last_gop)*1000/29.97)) );
+		}
+
+/*		if (dec_ctx->false_pict_header)
```
Comment (anshul1912) [inline src/ccextractor.c]: Why this in comment

**Comment by abhishek-vinjamoori** [comment]
@ccx-testerbot runtests master

**Comment by abhishek-vinjamoori** [comment]
Tests look good !

**Pull request merged**